model/src/cmd/ModelCommandAPI.cc

```diff
@@ -0,0 +1,369 @@
+    {
+      const std::string poseInfo =
+            "\n        [" + std::to_string(poseComp->Data().X()) + " | "
+                          + std::to_string(poseComp->Data().Y()) + " | "
+                          + std::to_string(poseComp->Data().Z()) + "]\n"
+              "        [" + std::to_string(poseComp->Data().Roll()) + " | "
+                          + std::to_string(poseComp->Data().Pitch()) + " | "
+                          + std::to_string(poseComp->Data().Yaw()) + "]";
+
+      std::cout << "    - Pose: " << poseInfo << std::endl;
+    }
+  }
+}
+
+//////////////////////////////////////////////////
+// \brief Print the model joints information.
+// \param[in] _entity Entity of the model requested.
+// \param[in] _ecm ECM ready for requests.
+// \param[in] _jointName Joint to be printed, if nullptr, print all joints.
+void printJoints(const uint64_t entity,
+                const ignition::gazebo::EntityComponentManager &_ecm,
+                const std::string &_jointName)
+{
+  static const std::map<sdf::JointType, std::string> jointTypes = {
+    {sdf::JointType::REVOLUTE, "revolute"},
+    {sdf::JointType::BALL, "ball"},
+    {sdf::JointType::CONTINUOUS, "continuous"},
+    {sdf::JointType::FIXED, "fixed"},
+    {sdf::JointType::GEARBOX, "gearbox"},
+    {sdf::JointType::PRISMATIC,  "prismatic"},
+    {sdf::JointType::REVOLUTE2, "revolute2"},
+    {sdf::JointType::SCREW, "screw"},
+    {sdf::JointType::UNIVERSAL, "universal"}
+  };
+
+  const auto joints = _ecm.EntitiesByComponents(
+  ignition::gazebo::components::ParentEntity(entity),
+  ignition::gazebo::components::Joint());
+
+  for(const auto &_entity : joints)
+  {
+    const auto parentComp =
+        _ecm.Component<ignition::gazebo::components::ParentEntity>(_entity);
+
+    const auto nameComp =
+        _ecm.Component<ignition::gazebo::components::Name>(_entity);
+
+    if(_jointName.length() && _jointName != nameComp->Data())
+        continue;
+
+    if (parentComp)
+    {
+      std::string parentInfo;
+      auto parentNameComp =
+          _ecm.Component<ignition::gazebo::components::Name>(
+          parentComp->Data());
+
+      if (parentNameComp)
+      {
+        parentInfo += parentNameComp->Data() + " ";
+      }
+      parentInfo += "[" + std::to_string(parentComp->Data()) + "]";
+
+      std::cout << "  - Joint [" << _entity << "]" << std::endl
+                << "    - Name: " << nameComp->Data() << std::endl
+                << "    - Parent: " << parentInfo << std::endl;
+    }
+
+    const auto jointComp =
+        _ecm.Component<ignition::gazebo::components::JointType>(_entity);
+    const auto childLinkComp =
+        _ecm.Component<ignition::gazebo::components::ChildLinkName>(_entity);
+    const auto parentLinkComp =
+        _ecm.Component<ignition::gazebo::components::ParentLinkName>(_entity);
+
+    if (childLinkComp && parentLinkComp)
+    {
+      std::cout << "    - Joint type:  " << jointTypes.at(jointComp->Data())
```

Comment:
I think the word "joint" can be omitted here because it's redundant.

I also noticed that the use of `[ ]` to wrap values in inconsistent. Some names have it, others don't... It would be nice to be consistent, either use it everywhere or nowhere.

Reply:
Addressed
I left all values and vector with `[ ]` and all the names without it